src/pkg/backup/details/details.go

```diff
@@ -864,15 +856,8 @@ func (i OneDriveInfo) Values() []string {
 	}
 }
 
-func (i *OneDriveInfo) UpdateParentPath(newPath path.Path, _ *path.Builder) error {
-	newParent, err := path.GetDriveFolderPath(newPath)
-	if err != nil {
-		return clues.Wrap(err, "making oneDrive path").With("path", newPath)
-	}
-
-	i.ParentPath = newParent
-
-	return nil
+func (i *OneDriveInfo) UpdateParentPath(newLocPath *path.Builder) {
+	i.ParentPath = newLocPath.PopFront().String()
```

Comment:
This makes me want for a well defined `path.Location` instead of a `path.Builder`.  Something with built-in knowledge of concepts like "this location has prefix of len 1 (alt, this location has a drive prefix), which is separate from its folders".  The builder is a good underlying data structure, but it isn't very good at communicating contextual expectations like this.  And I worry that there's footguns abound for the next contributor who doesn't have that topology already in mind.

Reply:
mmmm I think there's a broader question here about whether we need to continue providing `ParentPath` or not. Really this information is now rolled into `LocationRef`, but we've chosen different representations for `LocationRef` and `ParentPath` (hence this fun piece of code). Really this is a `path.Builder` that was obtained by calling `newLoc.InDetails()` :(